local_publisher.py

```diff
@@ -22,25 +22,39 @@ class LocalPublisher(TerminatorBlock):
 
     def __init__(self):
         super().__init__()
-        self._publisher = None
+        self._publishers = {}
 
     def configure(self, context):
         super().configure(context)
-        self._publisher = NioPublisher(
-            topic="{}.{}".format(self.local_identifier(), self.topic()))
-        self._publisher.open()
+        try:
+            publisher = NioPublisher(
+                topic="{}.{}".format(self.local_identifier(), self.topic()))
+            publisher.open()
+            self._publishers[topic] = publisher
+        except:
+            # topic is dynamic based on signals
+            pass
 
     def stop(self):
         """ Stop the block by closing the underlying publisher """
-        self._publisher.close()
+        for publisher in self._publishers.values():
+            publisher.close()
         super().stop()
 
     def process_signals(self, signals):
         """ Publish each list of signals """
+        topic = self.topic(signals[0])
```

Comment:
Are we ensuring somewhere that the topic will always be defined in the `signals[0]` signal?

Reply:
i.e. should this loop over the `signals` list?

Reply:
For an early implementation I and @hansmosh were thinking that it would be adequate to evaluate `topic` from the first signal in a list. The final desired behavior may be something more complex, e.g evaluate each signal's topic and construct a new list of signals for each topic and publish accordingly.